packages/@n8n/backend-core/src/decorators/__tests__/module.test.ts

```diff
@@ -0,0 +1,33 @@
+import { Container } from '@n8n/di';
+import { mock } from 'jest-mock-extended';
+import type { ExecutionLifecycleHooks } from 'n8n-core';
```

Comment:
perhaps we can define an interface in this package that `ExecutionLifecycleHooks` implements, and then remove the `n8n-core` dependency from this package?